jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -246,6 +246,31 @@ void java_add_components_to_class(
     components.end(), components_to_add.begin(), components_to_add.end());
 }
 
+void add_method_call_if_present(
```

Comment:
It's a bit weird to have a general-purpose utility function that only works on no-arg functions. I think you should take a vector-of-exprt and append those if given.